### DISCUSSION TITLE
Trash in pcmanfm-qt: Default behaviour with files from other devices/partitions

### DISCUSSION THREAD

**Discussion by buhtz** [lxqt/lxqt#2068] [General]
Before I open a "real" Bug Report I would like to understand some things.

I read https://github.com/lxqt/pcmanfm-qt/issues/1336 but did not fully understand.

Is it correct hat pcmanfm-qt should be able to handle deleting files from other partitions/devices?

Can you please describe the exact behaviour.

Is it done by default? Or is there a setting checkbox somewhere?

**Comment by tsujan** [comment]
I repeat my answer from here: https://github.com/lxqt/pcmanfm-qt/issues/1438#issuecomment-913646376

Trash is controlled by GVFS, not by pcmanfm-qt. The latter just will call the former if you have GVFS and checked Preferences → Behavior → File Operations → `Move deleted files "trash bin"...` and unchecked `Erase files on removable media...`. By default, the first option is checked and the second one is unchecked, so that trash is used when GVFS is installed.

Also see https://github.com/lxqt/pcmanfm-qt/wiki#permanently-deleting-files

**Comment by tsujan** [comment]
Also note that GVFS doesn't support Trash on all removable media. In such cases, pcmanfm-qt shows a prompt dialog.

In short, everything is controlled by GVFS.

**Comment by buhtz** [comment]
`GVFS` are simply four letters. Can you be more specific please.

**Reply by tsujan** [reply]
https://en.wikipedia.org/wiki/GVfs
